src/NosCore.GameObject/Character.cs

```diff
@@ -162,6 +189,153 @@ public int IsReputHero()
+
+                subpackets.Add(new FinitSubPacket
+                {
+                    CharacterId = relation.RelatedCharacterId,
+                    RelationType = relation.RelationType,
+                    IsOnline = ServerManager.Instance.IsCharacterConnected(relation.RelatedCharacterId),
+                    CharacterName = relation.CharacterName
+                });
+            }
+
+            return new FinitPacket { SubPackets = subpackets };
+        }
+
+        public void DeleteBlackList(long characterId)
+        {
+            CharacterRelationDTO relation = CharacterRelations.Values.FirstOrDefault(s => s.RelatedCharacterId == characterId && s.RelationType == CharacterRelationType.Blocked);
+
+            if (relation == null)
+            {
+                Session.SendPacket(new InfoPacket
+                {
+                    Message = Language.Instance.GetMessageFromKey(LanguageKey.CANT_FIND_CHARACTER, Session.Account.Language)
+                });
+                return;
+            }
+
+            CharacterRelations.TryRemove(relation.CharacterRelationId, out _);
+            Session.SendPacket(GenerateBlinit());
+        }
+
+        public void AddRelation(long characterId, CharacterRelationType relationType)
+        {
+            var relation = new CharacterRelation
+            {
+                CharacterId = CharacterId,
+                RelatedCharacterId = characterId,
+                RelationType = relationType,
+                CharacterName = ServerManager.Instance.Sessions.Values.FirstOrDefault(s => s.Character.CharacterId == characterId)?.Character.Name
+            };
+
+            CharacterRelations[relation.CharacterRelationId] = relation;
+
+            if (relationType == CharacterRelationType.Blocked)
+            {
+                Session.SendPacket(GenerateBlinit());
+                return;
+            }
+
+            Session.SendPacket(GenerateFinit());
+        }
+
+        public void DeleteRelation(long characterId)
+        {
+            CharacterRelationDTO characterRelation = CharacterRelations.Values.FirstOrDefault(s => s.RelatedCharacterId == characterId);
+            CharacterRelationDTO targetRelation = DAOFactory.CharacterRelationDAO.FirstOrDefault(s => s.CharacterId == characterId && s.RelatedCharacterId == CharacterId);
+
+            if (characterRelation == null || targetRelation == null)
+            {
+                return;
+            }
+
+            DAOFactory.CharacterRelationDAO.Delete(targetRelation.CharacterRelationId);
+            CharacterRelations.TryRemove(characterRelation.CharacterRelationId, out _);
+            Session.SendPacket(GenerateFinit());
+            //TODO: Refresh target relation if online, this is just for testing purposes
+            ClientSession target = ServerManager.Instance.Sessions.Values.FirstOrDefault(s => s.Character.CharacterId == characterId);
+            target?.SendPacket(target?.Character.GenerateFinit());
```

Comment:
you also need to remove the relation from this session. Moreover this step need to be before database deletion